### PR TITLE
CAP-196 search exception with postgresql -- move session init code to interceptor

### PR DESCRIPTION
**Pull request by bjagg**
Currently, user session information is populated by CalendarController. If another controller is hit, say from performing a search in portal, before calendar is visited, user session is not set.

Pulling the user session initialization code out from the CalendarController into an interceptor.

### CODE REVIEW

File: src/main/java/org/jasig/portlet/calendar/dao/HibernateCalendarSetDao.java [hunk 1]
```diff
@@ -54,18 +59,21 @@ public void setCalendarStore(CalendarStore calendarStore) {
...
         final Set<UserDefinedCalendarConfiguration> calendars = new HashSet<UserDefinedCalendarConfiguration>();
-        calendars.addAll(cals);
-        
-        final CalendarSet<UserDefinedCalendarConfiguration> set = new CalendarSet<UserDefinedCalendarConfiguration>();
-        set.setConfigurations(calendars);
+        final CalendarSet<UserDefinedCalendarConfiguration> set
+                = new CalendarSet<UserDefinedCalendarConfiguration>(calendars);
+
+        final String username = getUsername(request);
+        if (username != null) {
```
Comment (jameswennmacher) [inline src/main/java/org/jasig/portlet/calendar/dao/HibernateCalendarSetDao.java]: Should it log an error in this case?  Should username ever be non-null?

File: src/main/java/org/jasig/portlet/calendar/dao/HibernateCalendarSetDao.java [hunk 2]
```diff
@@ -74,19 +82,26 @@ public void setCalendarStore(CalendarStore calendarStore) {
 
         final String username = getUsername(request);
 
-        // For this ICalendarSetDao, the list is the unfiltered collection from CalendarStore...
-        return calendarStore.getPredefinedCalendarConfigurations(username, false);
+        if (username != null) {
+            // For this ICalendarSetDao, the list is the unfiltered collection from CalendarStore...
+            return calendarStore.getPredefinedCalendarConfigurations(username, false);
+        } else {
+            return new ArrayList<PredefinedCalendarConfiguration>();
```
Comment (jameswennmacher) [inline src/main/java/org/jasig/portlet/calendar/dao/HibernateCalendarSetDao.java]: Log as error?

File: src/main/java/org/jasig/portlet/calendar/mvc/UserSessionInitializer.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import java.util.List;
+
+
+/**
+ * {code PortletFilter} to run initialization code when a user session is first created.
+ *
+ * @author Benito J. Gonzalez <bgonzalez2@unicon.net>
+ * @since 3.0.1
+ */
+public class UserSessionInitializer implements HandlerInterceptor {
```
Comment (jameswennmacher) [inline src/main/java/org/jasig/portlet/calendar/mvc/UserSessionInitializer.java]: I think it's good having initialization all done in one place.

**Comment by blamonet** [comment]
@bjagg this seemed to work for us. We aren't getting the "bytea" stacktrace now.

File: src/main/java/org/jasig/portlet/calendar/dao/HibernateCalendarSetDao.java (hunk 1 above)
Comment (bjagg) [inline src/main/java/org/jasig/portlet/calendar/dao/HibernateCalendarSetDao.java]: At this point, the session init code should have set this to 'guest' if the user is unauthenticated. Yep, logging is warranted.

File: src/main/java/org/jasig/portlet/calendar/dao/HibernateCalendarSetDao.java (hunk 2 above)
Comment (bjagg) [inline src/main/java/org/jasig/portlet/calendar/dao/HibernateCalendarSetDao.java]: Will do

**Pull request merged**